### PR TITLE
Metric: Add LabelKey type that has key and description.

### PR DESCRIPTION
**Pull request by songy23**

### CODE REVIEW

File: opencensus/proto/stats/metrics/metrics.proto [hunk 1]
```diff
@@ -68,9 +68,17 @@ message MetricDescriptor {
   }
   Type type = 4;
 
-  // TODO: Consider to add LabelInfo{key, description} if needed.
```
Comment (g-easy) [inline opencensus/proto/stats/metrics/metrics.proto]: Why is description needed?

File: opencensus/proto/stats/metrics/metrics.proto [hunk 2]
```diff
@@ -68,9 +68,17 @@ message MetricDescriptor {
...
-  repeated string label_keys = 5;
+  repeated LabelKey label_keys = 5;
+}
+
+// Defines a label key associated with a metric descriptor.
+message LabelKey {
+  // The key for the label.
+  string key = 1;
+
+  // The description of this label key.
```
Comment (g-easy) [inline opencensus/proto/stats/metrics/metrics.proto]: Please elaborate. What is the description for? Do any current exporters use this, or is it just a comment for human readers?

Comment (songy23) [reply opencensus/proto/stats/metrics/metrics.proto]: It's an optional field for other monitoring backend (like Stackdriver Monitoring). For now our exporters need to hard-code the description field: [CPP](https://github.com/census-instrumentation/opencensus-cpp/blob/master/opencensus/exporters/stats/stackdriver/internal/stackdriver_utils.cc#L40), [Java](https://github.com/census-instrumentation/opencensus-java/blob/master/exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtils.java#L79).

In addition, I think adding description for label key could help human reader, especially for keys like rpc_client_method vs rpc_server_method. We also have descriptions in the [specs](https://github.com/census-instrumentation/opencensus-specs/blob/master/stats/gRPC.md#tags).

File: opencensus/proto/stats/metrics/metrics.proto (hunk 1 above)
Comment (g-easy) [reply opencensus/proto/stats/metrics/metrics.proto]: Answer: some backends make use of it?

File: opencensus/proto/stats/metrics/metrics.proto (hunk 2 above)
Comment (g-easy) [reply opencensus/proto/stats/metrics/metrics.proto]: How about something like "a human-readable description of what this labelkey represents"

File: opencensus/proto/stats/metrics/metrics.proto (hunk 1 above)
Comment (songy23) [reply opencensus/proto/stats/metrics/metrics.proto]: Yes, 1. it's useful for some backend, and 2. provides more information to human readers.

File: opencensus/proto/stats/metrics/metrics.proto (hunk 2 above)
Comment (songy23) [reply opencensus/proto/stats/metrics/metrics.proto]: Sounds good, done.

**Pull request merged**